pkgs/applications/misc/scli/default.nix

```diff
@@ -0,0 +1,24 @@
+{ lib, buildPythonApplication, fetchFromGitHub, signal-cli, urwid, urwid-readline, dbus }: buildPythonApplication rec {
+  pname = "scli";
+  version = "0.6.0";
+  src = fetchFromGitHub {
+    owner = "isamert";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "1q2p0i4ar5xlpj4nbhadj67k9r7m7ph2x15q8zlw9bch6lixv8rg";
+  };
+  propagatedBuildInputs = [ signal-cli urwid urwid-readline dbus ];
+  dontBuild = true;
+  doCheck = false;
```

Comment:
It might be nice to include a checkPhase of just running `scli --help` or something similar to ensure it works. Maybe you could test its ability to pickup `signal-cli` too, once you get that to work.

Reply:
Unfortunately scli tries to create a ~/.local/share/scli directory upon initial run, which nix freaks out about:
https://gist.github.com/alex-eyre/dd3a0a8d8c132715725d703bd9841dac
I don't know if there's anything we can do about that :/

Reply:
You could probably do something like this:
```
XDG_DATA_HOME=$(mktemp -d)
XDG_CONFIG_HOME=$(mktemp -d)
```

Reply:
Thanks for the help! I've implemented a really basic invocation test in the checkPhase, but I'm not sure what more testing we can do without somehow logging into signal 🤔
12e50ce0117902c60eb880aacc672de6e23d6b25